src/lang/cpp/stmt.rs

```diff
@@ -249,12 +237,12 @@ fn if_statement(if_stmt: &AST) -> Option<IfStmt> {
 fn switch_statement(switch_stmt: &AST) -> Option<SwitchExpr> {
     let cond = switch_stmt
         .find_child_by_type(&["condition_clause"])
-        .map(|cond| expression(cond))??;
+        .map(expression)??;
     let cases = switch_stmt
         .find_child_by_type(&["compound_statement"])
         .map(|switch_stmt| switch_stmt.children.iter())?
-        .map(|case| switch_case(case))
-        .flat_map(|case| case)
+        .map(switch_case)
+        .flatten()
```

Comment:
Could we replace this with `flat_map(switch_case)`? Or is there some logistics issue I'm missing that would make that fail?

Reply:
Yes, not sure why I didn't realize that when making these changes.

Reply:
OK, I'm seeing it in a few other places throughout the codebase; would you like me to go through and switch them all now, or put it off until later?